### PR TITLE
Replace print_r() with var_dump() to skip out of memory limits

### PR DESCRIPTION
**Pull request by jissereitsma**

### CODE REVIEW

File: src/Whoops/Util/TemplateHelper.php [hunk 1]
```diff
@@ -183,7 +183,7 @@ public function dump($value)
             return $output;
         }
 
-        return print_r($value, true);
+        return var_dump($value, true);
     }
```
Comment (staabm) [inline src/Whoops/Util/TemplateHelper.php]: var-dump has no 2nd parameter which makes it return the value as a string.

Also I dont think that var-dump uses different memory then any other funtction

File: src/Whoops/Util/TemplateHelper.php [hunk 2]
```diff
@@ -183,7 +183,7 @@ public function dump($value)
             return $output;
         }
 
-        return print_r($value, true);
+        return var_dump($value);
```
Comment (staabm) [inline src/Whoops/Util/TemplateHelper.php]: In case you want return the content from var-dump you need a output buffer to capture it

Comment (jissereitsma) [reply src/Whoops/Util/TemplateHelper.php]: I'm not sure where my mind was. `var_dump` should be `var_export`. Even though it works in Whoops, the signature of the method would not make sense.

File: src/Whoops/Util/TemplateHelper.php (hunk 1 above)
Comment (JefferyHus) [reply src/Whoops/Util/TemplateHelper.php]: Just wondering but isn't `var_export` fails in a recursive array?

Comment (jissereitsma) [reply src/Whoops/Util/TemplateHelper.php]: Not that I know of and I didn't find anything of this liking. The official PHP documentation mentions nothing of the liking. Some user comments refer to this, but those remarks seem to relate to older PHP versions. And that's precisely the issue here: `print_r` is known to misbehave. `var_export` and `var_dump` behave slightly better ...

**Comment by denis-sokolov** [comment]
I value the intent to improve this, thank you.

However, displaying data that is huge to cause memory issues is barely useful for the users, so that concern is not a priority in my opinion. And when choosing between `print_r` and `var_export`, I would rather rely on their meaning per the documentation: “human-readable information” vs “parsable string representation”, and here I choose human-readable information.

Perhaps you should disable the table output or use the Symfony dumper instead?

**Comment by jissereitsma** [comment]
@denis-sokolov I did not know the flexibility of Whoops yet. I simply added `symfony/var-dumper` to my own dependencies, and this bypasses the entire issue. Thanks!

**Pull request closed without merging**